fabric-network/test/impl/event/blocklistener.spec.ts

```diff
@@ -110,6 +113,40 @@ describe('block listener', () => {
 		});
 	}
 
+	function addTransaction(event:EventInfo, timestamp?:Date): EventInfo {
+		event.block.data = new fabproto6.common.BlockData();
+		event.block.data.data.push(newEnvelope(new fabproto6.protos.Transaction(), 'txn1', timestamp));
+		event.block.metadata = new fabproto6.common.BlockMetadata();
+		event.block.metadata.metadata = [];
+		event.block.metadata.metadata[fabproto6.common.BlockMetadataIndex.TRANSACTIONS_FILTER] = new Uint8Array(10);
+		return event;
+	}
+
+	function addFilteredTransaction(event: EventInfo, filteredTransaction: fabproto6.protos.FilteredTransaction): void {
+		event.filteredBlock.filtered_transactions.push(filteredTransaction);
+	}
+	function newFilteredTransaction(): fabproto6.protos.FilteredTransaction {
+		const filteredTransaction = new fabproto6.protos.FilteredTransaction();
+		filteredTransaction.tx_validation_code = fabproto6.protos.TxValidationCode.VALID;
+		filteredTransaction.transaction_actions = {};
+		return filteredTransaction;
+	}
+
+	function newEnvelope(transaction: any, transactionId?: string, timestamp?:Date): any {
+		const channelHeader :any = {};
+		channelHeader.type = fabproto6.common.HeaderType.ENDORSER_TRANSACTION;
+		channelHeader.tx_id = transactionId;
+		channelHeader.timestamp = timestamp.toISOString();
```

Comment:
Unsafe use of optional (`Date | undefined`) timestamp variable here, which would fail at runtime if not set